gui/app/services/updatesService.js

```diff
@@ -0,0 +1,89 @@
+(function(){
+  
+ //This handles updates
+ 
+ const _ = require('underscore')._;
+ const JsonDb = require('node-json-db');
+ const request = require('request');
+ const compareVersions = require('compare-versions');
+
+ angular
+  .module('firebotApp')
+  .factory('updatesService', function (settingsService) {
+    // factory/service object
+    var service = {}
+
+    // Update Checker
+    // This checks for updates.
+    service.updateCheck = function(){
+        return new Promise((resolve, reject) => {
```

Comment:
I recently learned how to make promises work better in Angular. You need to use Angular's promise implementation, which is called the $q service (odd name). Nice thing is you can just toss an existing Promise into it and it works (and then you wont need $scope.apply in the controller.

First inject $q into the updatesService
```javascript
.factory('updatesService', function (settingsService, $q) {
```
Then wrap the promise in $q.when();
```javscript
return $q.when(
    new Promise((resolve, reject) => {
        //logic 
}));
```
And now you should be able to remove the $scope.apply in the updates controller.